content/en/04-tutorials/01-pixie-101/07-kafka-monitoring.md

```diff
@@ -0,0 +1,182 @@
+> - Run `px demo interact px-kafka` and follow the directions to turn on the invoicing service delay.
+
+## Kafka Microservice Demo App
+
+The demo application you deployed in the [Prerequisites](#prerequisites) section has:
+
+- A single `order` topic.
+- One producer: the `order` service publishes messages to the `order` topic.
+- Two consumers: the `shipping` and `invoicing` services consume messages from the `order` topic. The `shipping` service is able to consume the order messages at the rate they are produced. The `invoicing` service is slower and unable to consume the order messages at the rate they are produced.
+
+::: div image-xl relative
+<svg title='' src='use-case-tutorials/kafka/kafka-microservice-app.png'/>
+:::
+
+To see the application's simple front end, navigate to the external IP (and port) of the `apache` service:
+
+```bash
+kubectl -n px-kafka get svc apache
+```
+
+To use the application, click on the Order page and scroll to the bottom of the page to select the `"Add Order"` button. Any orders added on the Order page should soon show up on both the Shipping and Invoicing pages.
+
+## Topic-centric Flow Graph
+
+Let's use Pixie to see a graph of producers and consumers for each Kafka topic.
+
+1. Select `px/kafka_overview` from the script drop-down menu.
+
+2. Select the drop-down arrow next to the `namespace` argument, type `px-kafka`, and press `Enter` to re-run the script.
+
+::: div image-xl relative
+<svg title='' src='use-case-tutorials/kafka/kafka-overview.png'/>
+:::
+
+> Pixie is able to automatically discover and list all clients and topics in the cluster.
+
+> This graph shows us that our microservice app has 1 topic, named `order`, with 1 producer and 2 consumers.
+
+3. Hover over an edge on the graph to see throughput and total record Bytes for a producer or consumer. The thickness of the edge indicates an increase in throughput.
+
+> By examining the edges of the graph, we can see that the producer is producing to the order topic at about `450 B/s`. The `shipping` client is consuming at the same rate of `450 B/s`. The `invoicing` client is only consuming at `200 B/s` and is falling behind. Since this is a demo application, the traffic throughput is lower than what you'd typically see in a production application.
+
+> The **Kafka Topics** table below the graph summarizes the same high-level topic information, and also includes the number of topic partitions.
+
+4. Scroll down the page to see tables listing the brokers, producer and consumer pods.
+
+::: div image-xl relative
+<svg title='' src='use-case-tutorials/kafka/kafka-overview-2.png'/>
+:::
+
+> The **Kafka Producer Pods** and **Kafka Consumer Pods** tables confirm that the `invoicing` client is consuming messages at a slower speed than they are being produced.
+
+> The `shipping` client has sent 292 `Fetch` requests while the `invoicing` client has sent much fewer (only 192 `Fetch` requests). This again indicates that something is wrong.
```

Comment:
"In our case, the `shipping` client has..."

to clarify that their numbers might be different.